### ISSUE TITLE
mock-get-request incompatible with returns: { models: ... } when the models have polymorphic: false

### ISSUE THREAD

**Issue by gleseur** [danielspaniel/ember-data-factory-guy#320]
Hi,

We ran into some issues with the following code:

```javascript
// Factory
import FactoryGuy from 'ember-data-factory-guy';
FactoryGuy.define('carrierBill', {
  polymorphic: false,
  default: {
    type: 'CarrierBill::Something',
  },
});

// Test code
const carrierBill = make('carrierBill');
const carrierBills = [carrierBill]
mockQuery('carrierBill', {
   limit: 50,
   offset: 0,
   'sort[id]': 'desc',
}).returns({ models: carrierBills });

// Later on
carrierBill.get('type')
// Expected output => 'CarrierBill::Something'
// Actual output => 'carrier-bill
```
After some debugging, I think this line https://github.com/danielspaniel/ember-data-factory-guy/blob/master/addon/mocks/mock-get-request.js#L103 is the reason for this

We changed our test so that it uses (maybe its not the best workaround)
```javascript
query = mockQuery(...)
query.returns({
  json: query.fixtureBuilder.convertForBuild(
    'carrierBill',
    carrierBills.map((bill) => { return { id: bill.get('id') } })
})
```

Basically not passing the type in the fixture converter.

Thanks

Ember 2.11.3
Ember data 2.12.2
Ember data factory guy 2.13.7 which is bit behind. I haven't tried the latest version, but given that this code is still the same, I assume the problem is still there.

**Comment by danielspaniel** [comment]
I guess what you are saying is this:
```javascript
const carrierBill = make('carrierBill');
carrierBill.get('type'); //=> carrier-bill 
```
Let me try and recreate that in a test, and if I can not, then I will ask you to do that ( in the FG repo ) and send the PR  so I can investigate.

 One thing to note is I would do things this way:
```javascript
FactoryGuy.define('carrier-bill', {
  ....
});
const carrierBill = make('carrier-bill');
```
since ember-data refers to models as 'carrier-bill' dasherized and not carrierBill . This is probably not the issue, but more of a way to have your setup be properly prepared ( in my opinion )

**Comment by danielspaniel** [comment]
Ok, that worked for me ( I can not see this issue ) , so you will have to recreate it and make a PR for me to check out, or give me access to your repo?

**Comment by gleseur** [comment]
Thanks @danielspaniel for looking into this.

No, that's not what I meant, it is really going through the mocked response that forces a new value for type after the call has been mocked. Just making is fine.

See https://github.com/danielspaniel/ember-data-factory-guy/blob/master/addon/mocks/mock-get-request.js#L103 =>  type: model.constructor.modelName` but I don't know the code enough to say why the type is forced here.

I'll try to add a failing test showing that.

**Comment by danielspaniel** [comment]
Ohhhhhhhh .. I get it .. let me check that again

**Comment by danielspaniel** [comment]
Sorry @gleseur , I can not make a failing test in the factory guy repo. if you can, let me know, or try latest factory guy? maybe that fixes it for some mysterious reason?

**Comment by gleseur** [comment]
@danielspaniel See https://github.com/danielspaniel/ember-data-factory-guy/pull/321 with a failing test and a possible fix.
Though I'm a bit doubtful about the fact that this is the rignt fix

**Comment by danielspaniel** [comment]
Thanks for the issue. That was a great bug and you found the solution. 
This is fixed now in  => v2.13.19

**Comment by gleseur** [comment]
Well, Thank you @danielspaniel  for your great work on this lib, we've been using it for a while now and it's been so useful in our test suite. I'm glad I was of some help.
